solidity/contracts/converter/BancorConverterUpgrader.sol

```diff
@@ -195,11 +199,21 @@ contract BancorConverterUpgrader is IBancorConverterUpgrader, ContractRegistryCl
             address connectorAddress = _oldConverter.connectorTokens(i);
             (virtualBalance, weight, isVirtualBalanceEnabled, , ) = _oldConverter.connectors(connectorAddress);
 
-            IERC20Token connectorToken = IERC20Token(connectorAddress);
-            _newConverter.addConnector(connectorToken, weight, isVirtualBalanceEnabled);
-
-            if (isVirtualBalanceEnabled)
-                _newConverter.updateConnector(connectorToken, weight, isVirtualBalanceEnabled, virtualBalance);
+            // Ether reserve
+            if (connectorAddress == address(0)) {
+                _newConverter.addETHReserve(weight);
+            }
+            // Ether reserve token
+            else if (connectorAddress == address(etherToken)) {
+                _newConverter.addETHReserve(weight);
+            }
+            // ERC20 reserve token
+            else {
+                IERC20Token connectorToken = IERC20Token(connectorAddress);

```

Comment:
virtual balance isn't handled for EtherToken / ETH - it should be a separate section - first add the reserve (and distinguish between ERC20, ETH and EtherToken, then add virtual balance if exists regardless of the reserve type

Reply:
@yudilevi : done.